autosklearn/data/validation.py

```diff
@@ -0,0 +1,297 @@
+        # Also, a numpy array will be created
+        y = sklearn.utils.check_array(
+            y,
+            force_all_finite=True,
+            accept_sparse='csr',
+            ensure_2d=False,
+        )
+        return y
+
+    def _validate_dataframe_features(
+        self,
+        X: pd.DataFrame,
+    ) -> Union[pd.DataFrame, np.ndarray]:
+        """
+        Interprets a Pandas
+        Uses .iloc as a safe way to deal with pandas object
+        """
+
+        # Start with the features
+
+        # Register if a column needs encoding
+        enc_columns = []
+
+        # Also, register the feature types for the estimator
+        feature_types = []
+
+        # Make sure each column is a valid type
+        for i, column in enumerate(X.columns):
+            if X[column].dtype.name in self.valid_pd_enc_dtypes:
+                enc_columns.append(i)
+                feature_types.append('categorical')
+            elif not np.issubdtype(X[column].dtype, np.number):
+                if X[column].dtype.name == 'Object':
+                    raise ValueError(
+                        "Input Column {} has invalid type object. "
+                        "Cast it to a numerical value before using it in Auto-Sklearn."
+                        "You can use pandas.to_numeric() to do this".format(
+                            column,
+                        )
+                    )
+                elif pd.core.dtypes.common.is_datetime_or_timedelta_dtype(
+                    X[column].dtype
+                ):
+                    raise ValueError(
+                        "Input Column {} has invalid time-type. "
+                        "Please convert the time information to a categorical value. "
+                        "Usually, time information can be separated into a set of categorical "
+                        "features. For example, hour of the day can be converted to 24 "
+                        " categories.".format(
+                            column,
+                        )
+                    )
+                else:
+                    raise ValueError(
+                        "Input Column {} has unsupported dtype {}. "
+                        "Supported column types are categorical/bool/numerical dtypes. "
+                        "Make sure your data is formatted in a correct way, "
+                        "before feeding it to Auto-Sklearn.".format(
+                            column,
+                            X[column].dtype.name,
+                        )
+                    )
+            else:
+                feature_types.append('numerical')
+
+        # Make sure we only set this once. It should not change
+        if not self.feature_types:
+            self.feature_types = feature_types
+
+        # This proc has to handle multiple calls, for X_train
+        # and X_test scenarios. We have to make sure also that
+        # data is consistent within calls
+        if enc_columns:
```

Comment:
I'm wondering whether there should also be an else statement checking that the new data only contains numerical attributes?

Reply:
Understood, would add a query to make sure the conversion works.
sklearn check array is called also to enforce this, but I can make the message more informative and for debug purposes on our end.